v3/src/components/graph/utilities/graph_utils.ts

```diff
@@ -74,31 +71,29 @@ export function computeNiceNumericBounds(min: number, max: number): { min: numbe
   return bounds
 }
 
-export function setNiceDomain(values: number[], scale: ScaleBaseType | undefined, axis: INumericAxisModel) {
-  const valueExtent = extent(values, d => d) as [number, number],
-    niceBounds = computeNiceNumericBounds(valueExtent[0], valueExtent[1])
-  axis.setTransitionDuration(1000)
-  axis.setDomain(niceBounds.min, niceBounds.max)
-  scale?.domain([niceBounds.min, niceBounds.max])  // We can't rely on useNumericAxis hook because of the transition
+export function setNiceDomain(values: (string | number)[], axisModel: IAxisModel) {
+  if (axisModel.isNumeric) {
+    const numericAxisModel = axisModel as INumericAxisModel,
+      valueExtent = extent(values, d => Number(d)) as unknown as [number, number],
+      niceBounds = computeNiceNumericBounds(valueExtent[0], valueExtent[1])
+    numericAxisModel.setDomain(niceBounds.min, niceBounds.max)
+  }
```

Comment:
Can be simplified to:
```suggestion
export function setNiceDomain(values: number[], axisModel: IAxisModel) {
  if (isNumericAxisModel(axisModel)) {
    const [minValue, maxValue] = extent(values, d => d) as [number, number],
      { min: niceMin, max: niceMax } = computeNiceNumericBounds(minValue, maxValue)
    axisModel.setDomain(niceMin, niceMax)
  }
```
making use of the type guard function that I added in a separate commit.